avocado/core/job.py

```diff
@@ -425,7 +425,37 @@ def _run(self):
 
         self.create_test_suite()
         self.pre_tests()
+        return self.run_tests()
 
+    def create_test_suite(self):
+        """
+        Creates the test suite for this Job
+
+        This is a public Job API as part of the documented Job phases
+        """
+        if (getattr(self.args, 'remote_hostname', False) and
+           getattr(self.args, 'remote_no_copy', False)):
+            self.test_suite = [(None, {})]
+        else:
+            try:
+                self.test_suite = self._make_test_suite(self.urls)
+            except loader.LoaderError as details:
+                stacktrace.log_exc_info(sys.exc_info(), 'avocado.app.debug')
+                self._remove_job_results()
+                raise exceptions.OptionValidationError(details)
+
+    def pre_tests(self):
```

Comment:
It'd be probably less confusing to put those methods in the final position before this patch, so it only contains the bits. Alternatively you could include intermittent patch which only reorganizes the methods as this is a bit ugly. (still not terrible, so I'd be able to merge it, but it should not be hard to split this patch)